lib/models/item.ts

```
@@ -0,0 +1,32 @@
+import { Data } from './data';
+import { DataStore } from './datastore';
+import { Link } from './link';
+import { Links } from './links';
+
+export interface ItemJSON {
+    href: string;
+    links: Link[];
+    data: Data[];
+}
+
+export interface ItemAPI {
+    link(rel: string): Link | undefined;
```

Result should not be undefined, it should throw an error when no link was found